conformance/lang/expressions/object-constructor-expr/object_methods.balt

```diff
@@ -0,0 +1,382 @@
+Test-Case: output
+Description: Test defining a method of an object.
+Labels: array-type, field-access-expr, int, list-constructor-expr, method-call-expr, method-defn, module-type-defn,
+        object-constructor-expr, object-field, object-type
+
+type ObjectType object {
+    int[] val;
+    function func1() returns int;
+};
+
+function init() {
+    ObjectType obj = object {
+        int[] val;
+
+        function init() {
+            self.val = [1, 2];
+        }
+
+        function func1() returns int {
+            return 10;
+        }
+    };
+    io:println(obj.val); // @output [1,2]
+    io:println(obj.func1()); // @output 10
+}
+
+Test-Case: error
+Description: Test defining mismatching method of an object at the object constructor.
+Labels: array-type, field-access-expr, int, list-constructor-expr, method-call-expr, method-defn, module-type-defn,
+        object-constructor-expr, object-field, object-type
+
+type ObjectType object {
+    int[] val;
+    function func1() returns int;
+};
+
+function errorFunction() {
+    ObjectType obj = object { // @error incompatible types: expected 'ObjectType', found 'object { int[] val; function func1 () returns (string); }'
+        int[] val;
+
+        function init() {
+            self.val = [1, 2];
+        }
+
+        function func1() returns string {
+            return "A";
+        }
+    };
+}
+
+Test-Case: output
+Description: Test defining methods of an object with meta data.
+Fail-Issue: ballerina-platform/ballerina-lang#34859
+Labels: annotation-decl, annotation, field-access-expr, method-defn, module-type-defn, object-constructor-expr,
+        record-type, string, type-cast-expr, typeof-expr, var
+
+type Annot record {
+    string a;
+};
+
+annotation Annot v1 on object function;
+
+function init() {
+    var obj = object {
+        # This is a document.
+        # It is for an object function.
+        @v1 {
+            a: "v1"
+        }
+        function f1() {
+        }
+    };
+
+    Annot annot = <Annot>(typeof obj.f1).@v1;
+    io:println(annot.a); // @output v1
+}
+
+Test-Case: output
+Description: Test defining methods of an object with visibility qualifiers.
+Labels: array-type, field-access-expr, int, list-constructor-expr, member-access-expr, method-call-expr, method-defn,
+        module-type-defn, object-constructor-expr, object-field, object-type
+
+function init() {
+    var obj =  object {
+        private int[] val = [1, 2];
+
+        private function func1() returns int {
+            return self.val[0];
+        }
+
+        public function func2() returns int {
+            return self.val[0];
+        }
+    };
+
+    int a = obj.func2();
+    io:println(a); // @output 1
+}
+
+Test-Case: error
+Description: Test accessing private method of an object.
+Fail-Issue: ballerina-platform/ballerina-lang#36417
```

Comment:
Added fail issue for https://github.com/ballerina-platform/ballerina-lang/issues/36417